### PR TITLE
Fix Constructor WeakMap requires 'new'

### LINKED ISSUE #14: ESLint causes build on windows to fail with default git settings
Because ESLINT is set to need linefeeds set to LF, by default on windows building this will fail.

There's a few possible workarounds - the easiest being setting global git config core.autocrlf to force linefeeds to LF. However, this can break other programs' builds. A local git config can also be used but that doesn't help if using NPM to pull this repo.

I'm going to do a pull request for adding a .gitattributes file which forces line feeds to LF regardless of the git config for *.js files in the repo.

### PR DESCRIPTION
**Pull request by Zhouzi**
When using this polyfill in Chrome, I'm getting this error:

> Constructor WeakMap requires 'new'

After some google searches, it seems that extending built-in is not supported. This PR adds a babel plugin allowing that.

Also fix #14 

EDIT: Chrome version is 60.0.3112.90, OS is macOS Sierra 10.12.5

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -33,8 +33,8 @@ module.exports = function(grunt) {
               match: /\{\/\* include\("(.*?)"\) \*\/\}/,
               replacement: (match, filename) => {
                 return grunt.file.read(filename)
-                            .replace(/\n$/, "")
-                            .replace(/^[^{]/gm, "    $&");
+                  .replace(/\n$/, "")
+                  .replace(/^[^{]/gm, "    $&");
```
Comment (Rob--W) [inline Gruntfile.js]: Could you undo the change to these two lines?

**Comment by Zhouzi** [comment]
@Rob--W I just reverted the changes to the mentioned lines.

**Comment by Rob--W** [comment]
Could you squash all commits too? Otherwise `git blame` will still unnecessarily point to these commits.

**Comment by Zhouzi** [comment]
Here you go, all in one 😄

**Comment by Rob--W** [comment]
Wait, the build script of webextension-polyfill itself does not transpile the source to ES5, so the `class DefaultWeakMap extends WeakMap` syntax should work without any issues in Chrome.

If we used a transpiler, then this patch would be good to go. However, we don't transpile, so why should this PR be merged?

**Comment by Zhouzi** [comment]
I double checked and in fact it works as expected as long as the code is not transpiled. The error comes from the transpiled version of the code (using babel). Which means the issue is not much of webextension-polyfill's business. Sorry for the mistake!
